src/test/java/com/capitalone/dashboard/event/TestResultEventListenerTest.java

```diff
@@ -0,0 +1,175 @@
+        setupData();
+        testResultEventListener.onAfterSave(new AfterSaveEvent<>(testResult, null, ""));
+        Performance performance = testResultEventListener.createPerformanceDoc(testResult,
+                getTestCapabilities().iterator().next(), perfCItem);
+        verify(performanceRepository, never()).save(performance);
+        assertThat(performance.getUrl(), is(testResult.getUrl()));
+        assertThat(performance.getType(), is(PerformanceType.ApplicationPerformance));
+    }
+
+    private CollectorItem getTestResultCollectorItem() {
+        CollectorItem collectorItem = new CollectorItem();
+        collectorItem.setId(ObjectId.get());
+        collectorItem.setCollector(new Collector("Test Collector", CollectorType.Test));
+        collectorItem.setNiceName("Test Collector");
+        collectorItem.getOptions().put("jobName", "TestResultEventListenerTest");
+        return collectorItem;
+    }
+
+    private TestResult getTestResult(CollectorItem collectorItem) {
+        TestResult testResult = new TestResult();
+        testResult.setId(ObjectId.get());
+        testResult.setType(TestSuiteType.Performance);
+        testResult.setCollectorItemId(collectorItem.getId());
+        testResult.setTestCapabilities(getTestCapabilities());
+        return testResult;
+    }
+
+    private Collector getPerfToolsCollector() {
+        Collector collector = new Collector(COLLECTOR_NAME, CollectorType.AppPerformance);
+        collector.setId(ObjectId.get());
+        return collector;
+    }
+
+    private void setupData() {
+        when(collectorItemRepository.findOne(Matchers.any(ObjectId.class))).thenReturn(getTestResultCollectorItem());
+        when(collectorRepository.save(Matchers.any(Collector.class))).thenReturn(getPerfToolsCollector());
+        when(collectorItemRepository.save(Matchers.any(CollectorItem.class))).thenReturn(getNewPerfCollectorItem(getPerfToolsCollector()));
+        when(performanceRepository.save(Matchers.any(Performance.class))).thenReturn(getPerformanceDoc());
+    }
+
+    public CollectorItem getNewPerfCollectorItem(Collector perfToolsCollector) {
+        CollectorItem newPerfCItem = new CollectorItem();
+        newPerfCItem.setId(ObjectId.get());
+        newPerfCItem.setCollector(perfToolsCollector);
+        newPerfCItem.setCollectorId(perfToolsCollector.getId());
+        return newPerfCItem;
+    }
+
+    public Collection<TestCapability> getTestCapabilities() {
+        TestCapability testCapability = new TestCapability();
+        testCapability.setDescription("TestResultEventListenerTest");
+        testCapability.setDuration(1L);
+
+        TestSuite suite = new TestSuite();
+        suite.setDescription("TestResultEventListenerTest");
+        suite.setDuration(1L);
+        suite.setType(TestSuiteType.Performance);
+
+        List<TestCase> testCases = new ArrayList<>();
+        TestCase testCase1 = new TestCase();
```

Comment:
Rename to be more descriptive
testCase1 to testCase_AvgResponseTime
testCaseStep1 to testCaseStep_TargetResponseTime
testCaseStep2 to testCaseStep_ActualResponseTime

repeat this naming pattern for the other 2 KPI's
This helps with readability and is emerging as naming best practice

Reply:
naming pattern followed for test files